server/src/main/java/org/elasticsearch/repositories/RepositoryData.java

```diff
@@ -184,6 +184,24 @@ protected RepositoryData copy() {
                 clusterUUID);
     }
 
+    /**
+     * Creates a copy of this instance that does not track any shard generations.
```

Comment:
This is kinda dumb, but necessary for BwC (tests) because we used to filter numeric generations when deserializing the cache which now would not be a thing anymore so I went with this to keep behavior exactly the same in this PR.

Reply:
> we used to filter numeric generations when deserializing the cache

Sorry I think I'm missing something here. We filter numeric generations when loading from the repo but we trusted the gens in the cache didn't we?

Throwing away _all_ the shard generations in the cached version seems different from what we do today too, although maybe I'm just not making a large enough logical leap.

Maybe more simply, could you point me to an example of a test that fails without calling this so I can poke around myself?